coremltools/converters/mil/frontend/torch/test/test_torch_ops.py

```diff
@@ -440,6 +440,26 @@ def test_avg_pool3d(
         )
         run_compare_torch(input_shape, model, backend=backend)
 
+class TestAdaptiveMaxPool:
+    @pytest.mark.parametrize(
+        "output_size, magnification, backend",
+        itertools.product(
+            [(1,1), (3,3),(6,6),(32,32)],
+            [1,2,4,5,6,7],
+            backends,
+        ),
+    )
+    def test_adaptive_max_pool2d(
+            self, output_size, magnification, backend
+    ):
+        # we maginify output_size to get input_size
+        # we only test adaptive pooling where input_size is a multiple of output_size
+        input_size = (magnification * output_size[0], magnification * output_size[1])
+        model = nn.AdaptiveMaxPool2d(
+            output_size
+        )
+        print("running compare torch")
+        run_compare_torch(input_size, model, backend=backend)
```

Comment:
Looks like the test case is not well defined. `input_size` should be 4D tensor for any pool2d op. Pasting the CI error here:
```
>           raise ValueError('Input dimension should be at least {}'.format(len(out_size) + 1))
375E           ValueError: Input dimension should be at least 3
```

Reply:
Yup. Just pushed a fix for the shape problem in the test, and I extended it to also include cases that are expected to fail given the current implementation for adaptive layers.

Reply:
Awesome. Thanks @algal!